nixos/modules/services/misc/mautrix-whatsapp.nix

```diff
@@ -0,0 +1,65 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mautrix-whatsapp;
+
+  configFile = pkgs.runCommand "mautrix-whatsapp" {
+    buildInputs = [ pkgs.mautrix-whatsapp pkgs.remarshal ];
+    preferLocalBuild = true;
+  } ''
+    mkdir -p $out
+    remarshal -if json -of yaml \
+      < ${pkgs.writeText "config.json" (builtins.toJSON cfg.configOptions)} \
+      > $out/config.yaml
+
+    ${pkgs.mautrix-whatsapp}/bin/mautrix-whatsapp -c $out/config.yaml -g -r $out/registration.yaml
+  '';
+
+in
+{
+  options.services.mautrix-whatsapp = {
+    enable = mkEnableOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable mautrix-whatsapp, a puppeting bridge between Matrix and WhatsApp.
+
+        See <link xlink:href="https://github.com/tulir/mautrix-whatsapp/wiki/Bridge-setup">here</link> for documentation.
+     '';
+    };
+
+    configOptions = mkOption {
+      type = types.attrs;
+      description = "This options will be transform in YAML configuration file for the bridge";
+    };
+
+    registrationPath = mkOption {
+      type = types.path;
+      default = "${configFile}/registration.yaml";
+      description = ''
+        This options will provide a path to the YAML registration file for matrix-synapse.
+        The registration.yaml file will be generated in the nix store, so you can refer to it with <literal>config.services.mautrix-whatsapp.registrationPath</literal>. This file must be added in the matrix-synapse service options <literal>services.matrix-synapse.app_service_config_files</literal>
```

Comment:
Dont' we want to do this automatically, at line 64?

Reply:
In my configuration.nix, I added this option to have the registrationPath in the matrix-synapse config.

app_service_config_files = [ config.services.mautrix-whatsapp.registrationPath ];

You see a better way to do this @flokli ?

Reply:
I'm not sure if I'm misunderstanding, but you can also set the `app_service_config_files` parameter in the `config` block.

Reply:
Sorry it's me who misunderstanding you suggestions.